instrumentation/net_http/lib/opentelemetry/instrumentation/net/http/patches/dup/instrumentation.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Net
+      module HTTP
+        module Patches
+          module Dup
+            # Module to prepend to Net::HTTP for instrumentation
+            module Instrumentation
+              USE_SSL_TO_SCHEME = { false => 'http', true => 'https' }.freeze
+
+              # Constant for the HTTP status range
+              HTTP_STATUS_SUCCESS_RANGE = (100..399)
+
+              def request(req, body = nil, &)
+                # Do not trace recursive call for starting the connection
+                return super unless started?
+
+                return super if untraced?
+
+                attributes = {
+                  OpenTelemetry::SemanticConventions::Trace::HTTP_METHOD => req.method,
+                  OpenTelemetry::SemanticConventions::Trace::HTTP_SCHEME => USE_SSL_TO_SCHEME[use_ssl?],
+                  OpenTelemetry::SemanticConventions::Trace::HTTP_TARGET => req.path,
+                  OpenTelemetry::SemanticConventions::Trace::NET_PEER_NAME => @address,
+                  OpenTelemetry::SemanticConventions::Trace::NET_PEER_PORT => @port,
+                  'http.request.method' => req.method,
+                  'url.scheme' => USE_SSL_TO_SCHEME[use_ssl?],
+                  'server.address' => @address,
+                  'server.port' => @port
+                }.merge!(OpenTelemetry::Common::HTTP::ClientContext.attributes)
+
+                path, query = split_path_and_query(req.path)
+                attributes['url.path'] = path unless attributes.key?('url.path')
```

Comment:
We check if `url.path` already exists to account for possible attributes previously set by a user. For example, `url.path' => 'REDACTED`